### ISSUE TITLE
ninja think empty depfile is a missing depfile

### ISSUE THREAD

**Issue by bobzhang** [ninja-build/ninja#1229]
It would be nice that `ninja` think empty depfile means there is no deps

**Comment by evmar** [comment]
Can you describe the use case more?  Are empty files produced by some
compiler?

**Comment by bobzhang** [comment]
Yes, our compiler will generate an empty file when the dependencies are none

**Comment by evmar** [comment]
Which compiler?

**Comment by bobzhang** [comment]
@evmar it is the compiler I am working on. https://github.com/bloomberg/bucklescript
I turned on such micro-optimization and came across a regression, so I rolled back

**Comment by jonesmz** [comment]
@bobzhang Would you be able to make a PR to fix this issue with Ninja?
